### ISSUE TITLE
Fix handling of initial `null` value

### ISSUE THREAD

**Issue by jamestalmage** [firebase/angularfire#540]
When a value is unset in firebase you end up binding the following to `$scope`:
```javascript
{$id:'...', $priority:'...', $value:null}
```
If you end up populating that location with an object (i.e. `{firstName:'James', lastName:'Talmage'}`), it should become:
```javascript
{$id:'...', $priority:'...', firstName:'James', lastName:'Talmage'}
```
but I'm still seeing the old `$value:null` property as well:
```javascript
{$id:'...', $priority:'...', $value:null, firstName:'James', lastName:'Talmage'}
```
~~and then there is an infinite loop of updates between angular and angularFire~~

**Comment by jamestalmage** [comment]
I think the infinite loop might have been my own doing, but it is definitely related to `$value:null` not getting cleaned up

**Comment by jamestalmage** [comment]
PR #541 has a test documenting what I am talking about

**Comment by jamestalmage** [comment]
Closing. Fixed in #541 which is now merged.
